infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutContextData.kt

```diff
@@ -0,0 +1,226 @@
+package fi.fta.geoviite.infra.tracklayout
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import fi.fta.geoviite.infra.common.*
+import fi.fta.geoviite.infra.common.DataType.STORED
+import fi.fta.geoviite.infra.common.DataType.TEMP
+import fi.fta.geoviite.infra.logging.Loggable
+
+// TODO: GVT-2426 This enum is an outdated concept that won't serve well with designs. It's only used by the front. Can we refactor it out?
+enum class ContextType {
+    OFFICIAL, NEW_DRAFT, EDITED_DRAFT, DESIGN, NEW_DESIGN_DRAFT, EDITED_DESIGN_DRAFT
+}
+
+interface LayoutContextAware<T> {
+    val id: DomainId<T>
+    val dataType: DataType
+    // TODO: GVT-2426 if we keep this, the field should be renamed
+//    val contextType: ContextType
+    val draftType: ContextType
+    @get:JsonIgnore val isDraft: Boolean get() = false
+    @get:JsonIgnore val isOfficial: Boolean get() = false
+    @get:JsonIgnore val isDesign: Boolean get() = false
+}
+
+sealed class LayoutConcept<T : LayoutConcept<T>>(contextData: LayoutContextData<T>) :
+    LayoutContextAware<T> by contextData, Loggable {
+
+    abstract val version: RowVersion<T>?
+    abstract val contextData: LayoutContextData<T>
+}
```

Comment:
Nyt kun näitä luen vielä uudestaan niin oikeastaan silläkin tälle voisi keksiä toisen termin että `Concept` on sanana todella lähellä `Context`:ia. Tätä filua lukiessa luin ensin about järjestäen kaikki `Context`:it `Concept`:eina